beater/beater.go

```diff
@@ -43,17 +45,21 @@ func (bt *beater) Run(b *beat.Beat) error {
 		go bt.client.PublishAll(events)
 	}
 
-	server := newServer(bt.config, callback)
-	start(server, bt.config.SSL)
-	defer stop(server)
+	bt.server = newServer(bt.config, callback)
+	bt.errCh = start(bt.server, bt.config.SSL)
 
 	logp.Info("apm-server is running! Hit CTRL-C to stop it.")
-	// Blocks until service is shut down
-	<-bt.done
 
-	return nil
+	err = <-bt.errCh
+	logp.Err(err.Error())
```

Comment:
This should probably also happen after line 56, as otherwise always an err is logged?